lightkurve/correctors/cbvcorrector.py

```diff
@@ -1,258 +1,946 @@
+            mission = 'Kepler'
+        elif HDU['PRIMARY'].header.__contains__('CAMPAIGN'):
+            mission = 'K2'
+        else:
+            raise Exception('This does not appear to be a Kepler or K2 FITS HDU')
+
+
+        extName = 'MODOUT_{0}_{1}'.format(module, output)
+
+        try:
+            # Read the columns and meta data
+            dataTbl = Table.read(HDU[extName], format="fits")
+            dataTbl.meta.update(HDU[0].header)
+            dataTbl.meta.update(HDU[extName].header)
+  
+            # TimeSeries-based objects require a dedicated time column
+            # Replace NaNs with default time '2000-01-01', otherwise,
+            # astropy.time.Time complains
+            nanHere = np.nonzero(np.isnan(dataTbl['TIME_MJD'].data))[0]
+            timeData = dataTbl['TIME_MJD'].data
+            timeData[nanHere] = Time(['2000-01-01'], scale='utc').mjd
+            cbvTime = Time(timeData, format='mjd')
+            dataTbl.remove_column('TIME_MJD')
+            
+            # Gaps are labelled as 'GAPFLAG' so rename!
+            dataTbl['GAP'] = dataTbl['GAPFLAG']
+            dataTbl.remove_column('GAPFLAG')
+
+            dataTbl.meta['MISSION'] = mission
+            dataTbl.meta['CBV_TYPE'] = 'SingleScale'
+            
+        except:
+            dataTbl = None
+            cbvTime = None
+
+        # Here we instantiate the actual object
+        return self(data=dataTbl, time=cbvTime, **kwargs)
 
-        if self.lc.mission == 'Kepler':
-            quarter = 'q{:02}'.format(self.lc.quarter)
+
+    @property
+    def mission(self):
+        return self.meta['MISSION']
+
+    @mission.setter
+    def mission(self, mission):
+        self.meta['MISSION'] = mission
+
+    @property
+    def cbv_type(self):
+        return self.meta['CBV_TYPE']
+
+    @cbv_type.setter
+    def cbv_type(self, cbv_type):
+        self.meta['CBV_TYPE'] = cbv_type
+
+    @property
+    def quarter(self):
+        if (self.mission == 'Kepler'):
+            return self.meta['QUARTER']
+        else:
+            return None
+
+    @quarter.setter
+    def quarter(self, quarter):
+        if (self.mission == 'Kepler'):
+            self.meta['QUARTER'] = quarter
+        else:
+            pass
+        
+    @property
+    def campaign(self):
+        if (self.mission == 'K2'):
+            return self.meta['CAMPAIGN']
+        else:
+            return None
+
+    @campaign.setter
+    def campaign(self, campaign):
+        if (self.mission == 'K2'):
+            self.meta['CAMPAIGN'] = campaign
+        else:
+            pass
+        
+    @property
+    def module(self):
+        return self.meta['MODULE']
+
+    @module.setter
+    def module(self, module):
+        self.meta['MODULE'] = module
+        
+    @property
+    def output(self):
+        return self.meta['OUTPUT']
+
+    @output.setter
+    def output(self, output):
+        self.meta['OUTPUT'] = output
+        
+    def __repr__(self):
```

Comment:
Time-permitting, I recommend implementing the custom repr by overriding `_base_repr_()` method instead.  This method is called by the repr methods of the `TimeSeries` parent class. You could copy/paste `LightCurve._base_repr_()` to see how it can be overriden.